### PR TITLE
Move to core InputStreamByteBody

### PR DESCRIPTION
**Pull request by yawkat**

### CODE REVIEW

File: servlet-core/src/main/java/io/micronaut/servlet/http/body/AvailableByteArrayBody.java [hunk 1]
```diff
@@ -32,8 +32,10 @@
  *
  * @author Jonas Konrad
  * @since 4.9.0
+ * @deprecated Use {@link io.micronaut.http.body.stream.AvailableByteArrayBody} from core instead
  */
 @Internal
+@Deprecated(forRemoval = true)
```
Comment (sdelamo) [inline servlet-core/src/main/java/io/micronaut/servlet/http/body/AvailableByteArrayBody.java:38]: I think you can remove this class. it was annotated as `@Internal`.

File: servlet-core/src/main/java/io/micronaut/servlet/http/body/InputStreamByteBody.java [hunk 2]
```diff
@@ -41,8 +41,10 @@
  *
  * @since 4.9.0
  * @author Jonas Konrad
+ * @deprecated Use {@link io.micronaut.http.body.stream.InputStreamByteBody} from core instead
  */
 @Internal
```
Comment (sdelamo) [inline servlet-core/src/main/java/io/micronaut/servlet/http/body/InputStreamByteBody.java:46]: I think you can remove this class. it was annotated as `@Internal`.

File: servlet-core/src/main/java/io/micronaut/servlet/http/ByteArrayBufferFactory.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import io.micronaut.core.io.buffer.ByteBufferFactory;
+
+/**
+ * {@link ByteBufferFactory} implementation based on simple byte arrays.
+ *
+ * @since 4.10.0
+ * @author Jonas Konrad
+ */
+@Internal
+public class ByteArrayBufferFactory implements ByteBufferFactory<Void, byte[]> {
```
Comment (sdelamo) [inline servlet-core/src/main/java/io/micronaut/servlet/http/ByteArrayBufferFactory.java:29]: should this be I core?

File: servlet-core/src/main/java/io/micronaut/servlet/http/body/AvailableByteArrayBody.java (hunk 1 above)
Comment (yawkat) [reply servlet-core/src/main/java/io/micronaut/servlet/http/body/AvailableByteArrayBody.java:38]: it is used by gcp and such, see the javadoc

**Review by sdelamo — APPROVED** [review]

File: servlet-core/src/main/java/io/micronaut/servlet/http/ByteArrayBufferFactory.java (hunk 3 above)
Comment (yawkat) [reply servlet-core/src/main/java/io/micronaut/servlet/http/ByteArrayBufferFactory.java:29]: Maybe eventually but not needed yet

**Pull request merged**